src/websocket.js

```diff
@@ -187,9 +187,9 @@ class WebSocket extends EventTarget {
 
     const client = proxyFactory(this);
     if (this.readyState === WebSocket.CONNECTING) {
-      failWebSocketConnection(client, code || CLOSE_CODES.CLOSE_ABNORMAL, reason);
+      failWebSocketConnection(client, code == null ? CLOSE_CODES.CLOSE_ABNORMAL : code, reason);
```

Comment:
code can not be `0` neither `null` because of the row 169

Reply:
Actually, because of line 168, `code` can be `undefined`. This line serves to default it to `CLOSE_CODES.CLOSE_ABNORMAL` if it is.

Reply:
Right, but undefined is a possible argument, it should be the same as 1000 code (https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/close#code)

But your change makes the opposite. It checks the `null` that is not possible by design and throws an error on the row 169

Reply:
I think this might be a misunderstanding. In JavaScript, `foo == null` (notice the double-equal-sign) is an idiomatic way to test for `foo` being `null` _or_ `undefined`. It is what `foo ?? bar` transpiles to ([example](https://esbuild.github.io/try/#dAAwLjI0LjAALS10YXJnZXQ9ZXM1AGZvbyA/PyBiYXI)).

Reply:
Oh, I see. I think the `==` operator should be forbidden by ESLint. Anyway, according to row 169, the code can either be `undefined` or a valid number, so this check is redundant.

At least, it can be replaced with an undefined check.